runtime/src/main/java/org/corfudb/runtime/collections/StreamingManager.java

```diff
@@ -0,0 +1,147 @@
+package org.corfudb.runtime.collections;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StreamSubscriptionException;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A streaming subscription manager that allows clients to listen on
+ * the transaction updates of interested tables. The updates will be
+ * streamlined and clients can get notifications via the registered
+ * call backs.
+ * <p>
+ * Created by WenbinZhu on 11/5/20.
+ */
+@Slf4j
+public class StreamingManager {
+
+    // Number of thread in polling and notification pool.
+    private static final int NUM_THREAD_PER_POOL = 4;
+
+    // Default buffer size for each subscription.
+    private static final int DEFAULT_BUFFER_SIZE = 50;
+
+    // Corfu runtime to interact with corfu streams.
+    private final CorfuRuntime runtime;
+
+    // A map of all stream listeners and their subscription contexts.
+    private final Map<StreamListener, StreamSubscription> subscriptions;
+
+    // Thread pool for executing stream polling tasks.
+    private final ExecutorService pollingExecutor;
+
+    // Thread pool for executing client call back tasks.
+    private final ExecutorService notificationExecutor;
```

Comment:
There are so many use cases where the client would want to control their own threading. I think we should just allow them to pass in an executor when calling subscribe. For example, they might need to prioritize some subscriptions over other subscriptions. 

Furthermore, it makes it harder to test. For example, the application can't write a deterministic unit test if they can't control the executors in StreamingManager. If the application is able to passs their own exeuctors, then they can deterministcly test call backs using something like  `MoreExecutors.newDirectExecutorService`.

Finally, if they own their threading then they have to manage the exeuctors lifecycle, which means we don't need to make a decision on their behalf on whether threads should be daemon or not, and its less code for us.

Reply:
I think the reason that we don't want users to pass the executor is that the executor is shared by all subscribers. Usually differnent subscribers are from different verticals, the executor serves them all, so it's not so appropriate for a vertical to specify an executor when subscribing. If any subscriber can specify any an executor, the resource usage can be a problem and we need to maintain a lot of exectutors from different subscribers which is kind of messy.

Reply:
Yes, that make sense, but the centralization shouldn't be hidden inside the library. The entity (i.e., application code) that is creating the StreamingManager should manage that.

Reply:
The entity that is creating CorfuStore is usually different from the veticals that are using it, so it's no better than we doing it. Also if we change that it will break a lot of application code, maybe later we can discuss with verticals to see how they are using it.

Reply:
That is my point. There needs to be a centralization point that figures out configuration before exposing it to verticals. That centralization point of figuring configuration shouldn't be managed by the library itself, but in upper layers. 

I guess there is two points here 1) what's a better design approach 2) what are the implications of this change (i.e., breaking existing code). I don't think that its a good idea to forgo #1 because of #2.